battle-service/src/main/java/com/company/battle/utils/services/UserAuthorizationService.java

```diff
@@ -0,0 +1,33 @@
+package com.company.battle.utils.services;
+
+import com.company.common.dtos.UserDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.*;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.server.ResponseStatusException;
+
+@Service
+@RequiredArgsConstructor
+public class UserAuthorizationService {
+
+    public UserDto getCurrentUser() {
+        final Jwt jwt = (Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
try to include userId into jwt. and get it from there. you don't need to call user service each time, due to it will bring significant overhead to the call stack(each request to battle sevice will rise request to user service)

Reply:
Hello, could you please explain a bit more about userId in jwt? In our case, auth-provider is google, and it's not clear how the battle-service can know anything about users without making a request to the user-service. So, do you want something like a login method in the user-service that takes the jwt we generate from the auth-provider (google), modifies it slightly, and adds the userId to it? What if we need more information from the user later, besides just id?